gammapy/makers/safe.py

```diff
@@ -247,6 +255,41 @@ def make_mask_energy_bkg_peak(dataset):
         energy_min = energy_axis.pix_to_coord(idx)
         return geom.energy_mask(energy_min=energy_min)
 
+    def make_mask_bkg_clip(self, dataset):
+        """Mask aberrant values in background maps
+
+        By default only non-finite values are masked.
+        If`n_95percentile` is not None then the values larger than 
+        `n_95percentile` times the 95% percentile value will also be masked.
+       
+ 
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Dataset to compute mask for.
+
+        Returns
+        -------
+        mask_safe : `~numpy.ndarray`
+            Safe data range mask.
+        """
+
+        bkg = dataset.background.data
+        mask = np.isfinite(bkg)
+        if self.n_95percentile is not None:
+            for k in range(bkg.shape[0]):
+                data = bkg[k, :, :]
+                thr = self.n_95percentile * np.nanpercentile(data[data > 0], 95)
+                mask[k, :, :] = np.abs(bkg[k, :, :]) < thr
+        if self.force_clipping:
+            dataset.background.data[~mask] = 0.0
+        if np.any(~mask):
+            bad_values = np.unique(bkg[~mask])
+            log.warning(f"Invalid values found in background masked \n {bad_values}")
+            # TODO: maybe we should store the bad values and their coord somewhere,
```

Comment:
As users explicitly request to fix the bad values, this might not be needed for now. If they are interested in the bad values, they could just keep them in the background map and check afterwards...

Reply:
ok I will remove this comment